src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,150 @@
+
+  public Shooter() {
+    SparkMaxUtils.initWithRetry(this::initSparks, Constants.SPARK_INIT_RETRY_ATTEMPTS);
+    pivotAngleMap = new InterpolatingDoubleTreeMap();
+    pivotAngleMap.put(Constants.PLACEHOLDER_DOUBLE, Constants.PLACEHOLDER_DOUBLE);
+    pivotAngleMap.put(Constants.PLACEHOLDER_DOUBLE, Constants.PLACEHOLDER_DOUBLE);
+  }
+
+  public void shoot() {
+    controllerA.setReference(ShooterConstants.SHOOTER_MOTOR_SPEED_RPM, ControlType.kVelocity);
+    controllerB.setReference(ShooterConstants.SHOOTER_MOTOR_SPEED_RPM, ControlType.kVelocity);
+  }
+
+  public void stop() {
+    controllerA.setReference(0, ControlType.kVelocity);
+    controllerB.setReference(0, ControlType.kVelocity);
+  }
+
+  /** Returns the cosine of the intake angle in degrees off of the horizontal. */
+  public double getCosineArmAngle() {
+    return Math.cos(
+        getPivotPosition() - ShooterConstants.POSITION_WHEN_HORIZONTAL_DEGREES);
+  }
+
+  public void controlPosition(double distance){
+    pivotDesiredPosition = pivotAngleMap.get(distance);
+    pivotController.setGoal(pivotDesiredPosition);
+    double timestamp = Timer.getFPGATimestamp();
+
+    double armDemandVoltsA = pivotController.calculate(getPivotPosition());
+    double armDemandVoltsB = pivotController.calculate(getPivotPosition());
+    if (!prevTimestamp.isEmpty()) {
+      armDemandVoltsB += ShooterCal.PIVOT_MOTOR_FF.calculate(prevVelocity, pivotController.getSetpoint().velocity, timestamp - prevTimestamp.get());
+    }
+    double armDemandVoltsC = ShooterCal.ARBITRARY_PIVOT_FEED_FORWARD_VOLTS * getCosineArmAngle();
+
+    prevTimestamp = Optional.of(timestamp);
+    prevVelocity = pivotController.getSetpoint().velocity;
+    
+    pivotMotor.setVoltage(armDemandVoltsA + armDemandVoltsB + armDemandVoltsC);
+  }
+
+
+/** Set arm to latching position (up = 180deg). Modified controlPosition. **/
+  public void setLatchPosition(){
+    pivotDesiredPosition = ShooterConstants.LATCH_ANGLE_DEGREES;
+    pivotController.setGoal(pivotDesiredPosition);   
+    double timestamp = Timer.getFPGATimestamp();
+
+    double armDemandVoltsA = pivotController.calculate(getPivotPosition());
+    double armDemandVoltsB = pivotController.calculate(getPivotPosition());
+    if (!prevTimestamp.isEmpty()) {
+      armDemandVoltsB += ShooterCal.PIVOT_MOTOR_FF.calculate(prevVelocity, pivotController.getSetpoint().velocity, timestamp - prevTimestamp.get());
+    }
+    double armDemandVoltsC = ShooterCal.ARBITRARY_PIVOT_FEED_FORWARD_VOLTS * getCosineArmAngle();
+
+    prevTimestamp = Optional.of(timestamp);
+    prevVelocity = pivotController.getSetpoint().velocity;
+    
+    pivotMotor.setVoltage(armDemandVoltsA + armDemandVoltsB + armDemandVoltsC);
+  }
```

Comment:
This seems like a lot of repeated code with `controlPosition`. Can you think of a way to reduce the repeated code?